Sources_v3/StreamChat/APIClient/Endpoints/Payloads/AttachmentPayload.swift

```diff
@@ -0,0 +1,159 @@
+        let author = try container.decodeIfPresent(String.self, forKey: .author)
+        self.author = author
+        var text = try container.decodeIfPresent(String.self, forKey: .text)?.trimmingCharacters(in: .whitespacesAndNewlines)
+        
+        title = (
+            try container.decodeIfPresent(String.self, forKey: .title)
+                ?? container.decodeIfPresent(String.self, forKey: .fallback)
+                ?? container.decodeIfPresent(String.self, forKey: .name)
+                ?? ""
+        ).trimmingCharacters(in: .whitespacesAndNewlines)
+        
+        // Parse Image URL.
+        imageURL = AttachmentPayload.fixedURL(
+            try container.decodeIfPresent(String.self, forKey: .image)
+                ?? container.decodeIfPresent(String.self, forKey: .imageURL)
+                ?? container.decodeIfPresent(String.self, forKey: .thumbURL)
+        )
+        
+        // Parse URL.
+        url = AttachmentPayload.fixedURL(
+            try container.decodeIfPresent(String.self, forKey: .assetURL)
+                ?? container.decodeIfPresent(String.self, forKey: .url)
+                ?? container.decodeIfPresent(String.self, forKey: .titleLink)
+                ?? container.decodeIfPresent(String.self, forKey: .ogURL)
+        )
+        
+        let type: AttachmentType
+        
+        if let typeString = try? container.decode(String.self, forKey: .type) {
+            let existsType = AttachmentType(rawValue: typeString)
+            
+            if existsType == .video {
+                if author == "GIPHY" {
+                    type = .giphy
+                    text = nil
+                } else if let url = url, url.absoluteString.contains("youtube") {
+                    type = .youtube
+                } else {
+                    type = existsType
+                }
+            } else {
+                type = existsType
+            }
+        } else if let _ = try? container.decodeIfPresent(String.self, forKey: .ogURL) {
+            // swiftlint:disable:previous unused_optional_binding
+            type = .link
+        } else {
+            type = .unknown
+        }
+        
+        self.type = type
+        self.text = text
+        file = (type == .file || type == .video) ? try AttachmentFile(from: decoder) : nil
+        actions = try container.decodeIfPresent([AttachmentAction].self, forKey: .actions) ?? []
+        extraData = try .init(from: decoder)
+    }
+    
+    private static func fixedURL(_ urlString: String?) -> URL? {
+        guard let string = urlString else {
+            return nil
+        }
+        
+        var urlString = string
+        
+        if urlString.hasPrefix("//") {
+            urlString = "https:\(urlString)"
+        }
+        
+        if !urlString.lowercased().hasPrefix("http") {
+            urlString = "https://\(urlString)"
+        }
+        
+        return URL(string: urlString)
+    }
```

Comment:
Any idea what this does? 😅

Reply:
Seems like links are coming in different format from backend 🤷‍♂️
Added comments to this method.